### PR TITLE
Fix s390x db/asm/x86_64

### PR DESCRIPTION
**Pull request by eagleoflqj**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

#16168. Fixed 6 XXs.

**Test plan**

```sh
liumeo@liumeo-s390x:~/radare2/test$ r2r db/asm/x86_64
Running from /home/liumeo/radare2/test
Loaded 1077 tests.
Skipping json tests because jq is not available.
[1077/1077]              1053 OK        24 BR        0 XX        0 FX
Finished in 2 minutes and 1 seconds.
```

**Closing issues**

...

### CODE REVIEW

File: libr/asm/p/asm_x86_nz.c [hunk 1]
```diff
@@ -5158,10 +5159,11 @@ static int assemble(RAsm *a, RAsmOp *ao, const char *str) {
 		if (!r_str_casecmp (instr.mnemonic, lt_ptr->mnemonic)) {
 			if (lt_ptr->opcode > 0) {
 				if (!lt_ptr->only_x32 || a->bits != 64) {
-					ut8 *ptr = (ut8 *)&lt_ptr->opcode;
-					int i = 0;
-					for (; i < lt_ptr->size; i++) {
-						data[i] = ptr[lt_ptr->size - (i + 1)];
+					ut64 opcode = lt_ptr->opcode;
+					int i = lt_ptr->size - 1;
```
Comment (trufae) [inline libr/asm/p/asm_x86_nz.c:5163]: Sizet

File: libr/asm/p/asm_x86_nz.c [hunk 2]
```diff
@@ -5113,10 +5113,11 @@ static int oprep(RAsm *a, ut8 *data, const Opcode *op) {
 					free (instr.mnemonic);
 					return -1;
 				}
-				ut8 *ptr = (ut8 *)&lt_ptr->opcode;
-				int i = 0;
-				for (; i < lt_ptr->size; i++) {
-					data[i + l] = ptr[lt_ptr->size - (i + 1)];
+				ut64 opcode = lt_ptr->opcode;
+				int i = lt_ptr->size - 1;
```
Comment (trufae) [inline libr/asm/p/asm_x86_nz.c:5117]: Sizet. Shifting on signed vars is undefined behaviour

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/asm/p/asm_x86_nz.c (hunk 1 above)
Comment (eagleoflqj) [reply libr/asm/p/asm_x86_nz.c:5163]: https://github.com/radareorg/radare2/blob/c83d0e0c5f96aa8b01ca85b8df035c59452ed78d/libr/asm/p/asm_x86_nz.c#L4204-L4210
opcode is defined ut64. size_t may cause problem on arm32, windows...

File: libr/asm/p/asm_x86_nz.c (hunk 2 above)
Comment (eagleoflqj) [reply libr/asm/p/asm_x86_nz.c:5117]: It's an unsigned variable, no undefined behavior I suppose.

File: libr/asm/p/asm_x86_nz.c (hunk 1 above)
Comment (trufae) [reply libr/asm/p/asm_x86_nz.c:5163]: The comment is in line 5163, not 5163. I said nothing to the opcode

File: libr/asm/p/asm_x86_nz.c (hunk 2 above)
Comment (trufae) [reply libr/asm/p/asm_x86_nz.c:5117]: Int is signed

File: libr/asm/p/asm_x86_nz.c (hunk 1 above)
Comment (eagleoflqj) [reply libr/asm/p/asm_x86_nz.c:5163]: lookuptable’s size field is int, why let i be size_t?

File: libr/asm/p/asm_x86_nz.c (hunk 2 above)
Comment (eagleoflqj) [reply libr/asm/p/asm_x86_nz.c:5117]: yes, but the shift isn’t operated on i.

Comment (wargio) [reply libr/asm/p/asm_x86_nz.c:5117]: i do not see anything wrong with this. shift occurs only on `opcode` which is unsigned64.

**Review by wargio — COMMENTED** [review]
i do not see any issue with the checks. `i` can't be negative

**Review by wargio — APPROVED** [review]

**Review by trufae — APPROVED** [review]

**Pull request merged**